sdk/ml/azure-ai-ml/azure/ai/ml/operations/_workspace_operations_base.py

```diff
@@ -654,44 +654,67 @@ def _populate_arm_parameters(self, workspace: Workspace, **kwargs: Dict) -> Tupl
-                else "",
-            )
-            _set_val(
-                param["online_store_connection_name"],
-                workspace._feature_store_settings.online_store_connection_name
-                if workspace._feature_store_settings.online_store_connection_name
-                else "",
-            )
+            if workspace._feature_store_settings.offline_store_connection_name:
+                _set_val(
+                    param["offline_store_connection_name"],
+                    workspace._feature_store_settings.offline_store_connection_name,
+                )
+            if workspace._feature_store_settings.online_store_connection_name:
+                _set_val(
+                    param["online_store_connection_name"],
+                    workspace._feature_store_settings.online_store_connection_name,
+                )
 
         if workspace._kind and workspace._kind.lower() == "featurestore":
             materialization_identity = kwargs.get("materialization_identity", None)
             offline_store_target = kwargs.get("offline_store_target", None)
             online_store_target = kwargs.get("online_store_target", None)
 
-            from azure.ai.ml._utils._arm_id_utils import AzureResourceId
+            from azure.ai.ml._utils._arm_id_utils import AzureResourceId, AzureStorageContainerResourceId
 
-            if offline_store_target is not None:
-                arm_id = AzureResourceId(offline_store_target)
-                _set_val(param["offline_store_target"], offline_store_target)
+            if offline_store_target:
+                arm_id = AzureStorageContainerResourceId(offline_store_target)
+                _set_val(param["offlineStoreStorageAccountOption"], "existing")
+                _set_val(param["offline_store_container_name"], arm_id.container)
+                _set_val(param["offline_store_storage_account_name"], arm_id.storage_account)
                 _set_val(param["offline_store_resource_group_name"], arm_id.resource_group_name)
                 _set_val(param["offline_store_subscription_id"], arm_id.subscription_id)
-            if online_store_target is not None:
+            else:
+                _set_val(param["offlineStoreStorageAccountOption"], "new")
+                _set_val(
+                    param["offline_store_container_name"],
+                    _generate_storage_container(workspace.name, resources_being_deployed),
+                )
+                if not workspace.storage_account:
+                    _set_val(param["offline_store_storage_account_name"], param["storageAccountName"]["value"])
+                else:
+                    _set_val(
+                        param["offline_store_storage_account_name"],
+                        _generate_storage(workspace.name, resources_being_deployed),
+                    )
+                _set_val(param["offline_store_resource_group_name"], workspace.resource_group)
```

Comment:
here, the offline store can still be a user provided storage account (that the user provide as the workspace default storage account), is that correct?  And is it possible that the sub and resource group of the storage account is not the same as the workspace?

Reply:
yes, user can provide storage account and offline store storage account, they can be the same storage account. In this case we don't provision any storage account for user because they are all provided. Role assignment will not be bothered as we are already able to handle cross sub/resource group for offline store.

Reply:
clarified with Runhan offline. I can understand the behavior now